packages/expect/src/jestMatchersObject.ts

```diff
@@ -56,57 +56,63 @@ export const setMatchers = (
 ): void => {
   Object.keys(matchers).forEach(key => {
     const matcher = matchers[key];
-    Object.defineProperty(matcher, INTERNAL_MATCHER_FLAG, {
-      value: isInternal,
-    });
-
-    if (!isInternal) {
-      // expect is defined
-
-      class CustomMatcher extends AsymmetricMatcher<
-        [unknown, ...Array<unknown>]
-      > {
-        constructor(inverse = false, ...sample: [unknown, ...Array<unknown>]) {
-          super(sample, inverse);
-        }
-
-        asymmetricMatch(other: unknown) {
-          const {pass} = matcher.call(
-            this.getMatcherContext(),
-            other,
-            ...this.sample,
-          ) as SyncExpectationResult;
 
-          return this.inverse ? !pass : pass;
-        }
-
-        toString() {
-          return `${this.inverse ? 'not.' : ''}${key}`;
-        }
+    if (matcher) {
```

Comment:
can we flip the condition. and do an early `return` to avoid the nesting?

Reply:
Of course.  Not sure what I was thinking...